File: jumpingJimmy2.cpp
int jumpingJimmy2(std::vector<int> tower, std::vector<int> power, std::vector<int> poison, int jumpHeight) {
    int totalHeight = 0;
    int tmp = 0;
    
    for(int i = 0;i< tower.size();i++){
        if(tmp +  tower.at(i) >= jumpHeight){
            
            if(tmp + tower.at(i) == jumpHeight){
                totalHeight += tower.at(i);
                tmp = 0;
                
                if(std::find(power.begin(), power.end(), i) != power.end()){
                    jumpHeight +=1;
                }else if (std::find(poison.begin(), poison.end(), i) != poison.end()){
                    jumpHeight -=1;
                }
            }else{
                if(std::find(power.begin(), power.end(), i-1) != power.end()){
                    jumpHeight +=1;
                }else if (std::find(poison.begin(), poison.end(), i-1) != poison.end()){
                    jumpHeight -=1;
                }
                
                tmp = tower.at(i);
                if(tmp > jumpHeight) break;
                
                totalHeight += tower.at(i);
                
            }
        }else{
            tmp += tower.at(i);
            totalHeight += tower.at(i);
        }
    }
    
    return totalHeight;
}

File: jumpingJimmy2.py
def jumpingJimmy2(tower, power, poison, jumpHeight):
	totalHeight = 0
	tmp = 0

	for i in range(len(tower)):

		if tmp + tower[i] >= jumpHeight:

			if tmp + tower[i] == jumpHeight:
				totalHeight += tower[i]
				tmp = 0

				if i in power:
					jumpHeight +=1
				elif i in poison:
					jumpHeight -=1
			else:

				if i-1 in power:
					jumpHeight +=1
				elif i-1 in poison:
					jumpHeight-=1

				tmp = tower[i]
				if tmp > jumpHeight:
					break

				totalHeight += tower[i]
		else:

			tmp += tower[i]
			totalHeight += tower[i]
	return totalHeight


